### DISCUSSION TITLE
Cache nas rotas

### DISCUSSION THREAD

**Discussion by RonaldoBueno16** [balta-io/forum#537] [Tira Dúvidas] [answered]
Boa noite!

Estou fazendo um projeto pessoal onde eu estou utilizando a seguinte arquitetura:
![image](https://github.com/balta-io/forum/assets/44504078/b92a0720-dd95-4c14-854b-24cb666afa83)

Tenho a seguinte rota na camada de API:
![image](https://github.com/balta-io/forum/assets/44504078/15441ea6-a518-4e57-9853-55f22f1892f5)
que está chamando o meu appserviço na camada de aplicação:
![image](https://github.com/balta-io/forum/assets/44504078/59651be0-c988-40d8-9e9a-d3a9fcd4d780)
que está chamando o meu serviço de listar produtos na camada de dominio:
![image](https://github.com/balta-io/forum/assets/44504078/ad6a8b96-3751-4d98-83d8-1541feee2ecd)
que está chamando o meu listarprodutos na camada de infraestrutura:
![image](https://github.com/balta-io/forum/assets/44504078/f080437b-59c7-4bec-935b-0a85c903106a)

O cache ele está implementado na camada da API conforme a primeira imagem, porém em um arquitetura limpa, é la que ele deve ficar?

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @RonaldoBueno16, como vai?

É uma opção... no caso você está usando um cache local, na API, então sim, é comum que ele fique ai.
Porém, neste cenário, cada máquina da sua API que subir, ele vai precisar "criar" este cache.

Outra opção é utilizar um cache distribuído, com Redis por exemplo!
